GameServer/Hubs/CaveHub.cs

```diff
@@ -17,13 +20,71 @@ public async Task SendCavernSound(ICommand command, Room room)
 
         public override async Task OnConnectedAsync()
         {
-            var welcomeMessage = "Welcome to the game!";
-            var commands = new List<string> { "Exit", "WalkToCommand" };
-
+            string welcomeMessage = "Welcome to the game!";
             await Clients.Caller.SendAsync("ReceiveWelcomeMessage", welcomeMessage);
-            await Clients.Caller.SendAsync("ReceiveAvailableCommands", commands);
+
+            await SendAvailableCommands();
 
             await base.OnConnectedAsync();
         }
+
+        public async Task ProcessPlayerCommand(string command)
+        {
+            Console.WriteLine($"Received command: {command}");
+
+            GameManager gameManager = GameManager.GetInstance;
+            Player player = gameManager.GetPlayer();
+            IRoom? room;
+
+            if (command.StartsWith("Walk to ", StringComparison.Ordinal))
+            {
+                string destination = command.Substring(8);
+                room = gameManager.GetRoom(destination);
+
+                if (room == null)
+                {
+                    await Clients.Caller.SendAsync("CommandProcessed", "Invalid command.");
+                    return;
+                }
+
+                var commandFactory = new CommandFactory();
+                ICommand commandToExecute = commandFactory.CreateCommand<WalkToCommand, IRoom?>(room);
+
+                if (commandToExecute != null)
+                {
+                    commandToExecute.Execute(player);
+
+                    await SendAvailableCommands();
+                }
+            }
+            else
+            {
+                await Clients.Caller.SendAsync("CommandProcessed", "Invalid command.");
+            }
+        }
+
+        // Helpers
+        private IReadOnlyList<IRoom> UpdateValidWalkToRooms()
+        {
+            GameManager gameManager = GameManager.GetInstance;
+            Player player = gameManager.GetPlayer();
+            IRoom currentRoom = player.GetCurrentRoom();
+
+            return currentRoom.GetConnectedRooms();
+        }
+
+        private async Task SendAvailableCommands()
```

Comment:
Should probably use the CommandRepository here, but for the moment ConsoleClients and GameServer projects use different Commands.
I didn't want to add dependencies between projects, so I let like this until we find a better solution to Serialize/Deserialize the sent message.